PyPDF2/_reader.py

```diff
@@ -254,8 +254,26 @@ def __init__(
         self.stream = stream
 
         self._override_encryption = False
-        if password is not None and self.decrypt(password) == 0:
-            raise PdfReadError("Wrong password")
+        self.encryption: Optional[Encryption] = None
```

Comment:
Consistently with my question in _merger, I would prefer if this was the private attribute `_encryption` instead of the public attribute `encryption. Except, of course, if there is a reason why users would want to access it.